### ISSUE TITLE
Wishlist breaks on PHP 5.4 and below

### ISSUE THREAD

**Issue by evensis** [dotmailer/dotmailer-magento-extension#291]
Hi,

Having the following issue with the wishlist output for dynamic content:

Fatal error: Can't use method return value in write context in <obfuscated>/app/code/community/Dotdigitalgroup/Email/Block/Wishlist.php on line 19

Looking at the code I can the issue is coming from: 

```
if ($wishlist && !empty($wishlist->getItemCollection())) {
```
Prior to PHP5.5, PHP does not support expressions within the scope of empty().

The second conditional will always return true, as empty() is essentially just an alias of isset() which looks for existence.

```if($wishlist && !empty($wishlist->getItemCollection())```

is logically equivalent to:

```if($wishlist && $wishlist->getItemCollection())```

I have changed this to:

```
if ($wishlist && count($wishlist->getItemCollection()) > 0) {
```
Which resolves the issue with the conditional, along with the crash associated with it.

We are using 6.3.9, just updated last week.

Thanks.

**Comment by adeelq** [comment]
@evensis Thanks for reporting this. Indeed with 5.4 there is an issue. We are resolving this issue and will be out in next release.

**Comment by cdiacon** [comment]
Fixed in 6.4.0
